packages/autocomplete/src/elements/Autocomplete.js

```diff
@@ -0,0 +1,305 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { MenuView, Item } from '@zendeskgarden/react-menus';
+import { TextGroup, Label, FauxInput, Input, Hint } from '@zendeskgarden/react-textfields';
+import { FieldContainer, KEY_CODES } from '@zendeskgarden/react-selection';
+import AutocompleteContainer from '../containers/AutocompleteContainer';
+
+const VALIDATION = {
+  SUCCESS: 'success',
+  WARNING: 'warning',
+  ERROR: 'error',
+  NONE: 'none'
+};
+
+const StyledNoItemsMessage = styled.div`
+  margin: 16px;
+  text-align: center;
+`;
+
+const StyledMenuOverflow = styled.div`
+  overflow-y: auto;
+  max-height: ${props => props.maxHeight || 'inherit'};
+`;
+
+const StyledInput = styled(Input)`
+  && {
+    display: inline-block;
+    ${props => !props.isOpen && 'width: 1px;'}
+  }
+`;
+
+const StyledFauxInput = styled(FauxInput)`
+  cursor: text;
+`;
+
+const StyledValueWrapper = styled.div`
+  display: inline-block;
+  vertical-align: middle;
```

Comment:
this component needs a `margin-left: -1px` (reversed for RTL) in order to prevent the visual shift that is occurring.